File: task_equivalents/llm_translations/gpt/base/r/data_manipulation_preprocessing/01_impute_encode_scale.R
library(readr)
library(dplyr)

Auto <- read_csv("data/csv/Auto.csv", na = "?") %>%
na.omit()

Auto <- Auto %>%
mutate(mpg01 = ifelse(mpg > median(mpg), 1, 0)) %>%
select(-mpg, -name)

var_no_scale <- c("cylinders", "year", "origin", "mpg01")
var_to_scale <- c("displacement", "horsepower", "weight", "acceleration")

scaled_var <- scale(Auto[, var_to_scale])

df <- bind_cols(
as.data.frame(scaled_var) %>% setNames(var_to_scale),
Auto[, var_no_scale]
)

x <- df[, 1:(ncol(df) - 1)]
y <- df$mpg01

File: task_equivalents/llm_translations/gpt/base/python/data_manipulation_preprocessing/01_impute_encode_scale.py
import pandas as pd
from sklearn.preprocessing import StandardScaler

Auto = pd.read_csv('data/csv/Auto.csv', header=0, na_values='?', index_col=0)
Auto = Auto.dropna()

Auto['mpg01'] = (Auto['mpg'] > Auto['mpg'].median()).astype(int)
Auto = Auto.drop(columns=['mpg', 'name'])

# Scale continuous variables; keep binary/categorical-coded columns unscaled.
var_no_scale = ['cylinders', 'year', 'origin', 'mpg01']
var_to_scale = ['displacement', 'horsepower', 'weight', 'acceleration']

scaler = StandardScaler()
scaled_var = pd.DataFrame(
    scaler.fit_transform(Auto[var_to_scale]),
    columns=var_to_scale,
    index=Auto.index
)

df = pd.concat([scaled_var, Auto[var_no_scale]], axis=1)

x = df.iloc[:, :-1]
y = df['mpg01']